gitlab/.gitlab-ci.yml

```diff
@@ -0,0 +1,417 @@
+    - rm -rf openxt-workspace || true
+    - mkdir -p openxt-workspace
+    - cd openxt-workspace
+    # Set up the sstate
+    - if [ ! -d /cache/sstate/$CI_COMMIT_REF_NAME ]; then mkdir /cache/sstate/${CI_COMMIT_REF_NAME}; fi
+    - ln -s /cache/sstate/${CI_COMMIT_REF_NAME} sstate-cache
+    # Download the sources for building
+    - repo init -u https://${MANIFEST_REPO} -b $MANIFEST_BRANCH --no-clone-bundle -m $MANIFEST_FILE
+    - repo sync --no-clone-bundle
+    - repo start ${CI_COMMIT_REF_NAME} openxt/*
+    # Configure the build
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} config -t stable-9 -b ${CI_COMMIT_REF_NAME}
+    # Copy the certs into the workspace
+    - mkdir -p certs && cp -R /certs/* certs/
+    # Update the mirrors
+    - cd build-${CI_COMMIT_REF_NAME}/conf
+    - awk '1;/MIRRORS .*" \\/{print "    http://v3.sk/~lkundrak/dev86/archive/             https://openxt.ainfosec.com/mirror/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+    - awk '1;/MIRRORS .*" \\/{print "    https://cgit.freedesktop.org/~airlied/vbetool/snapshot/             https://openxt.ainfosec.com/mirror/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+    - awk '1;/MIRRORS .*" \\/{print "    http://www.ivmaisoft.com/_bin/atomic_ops/             https://ftp.osuosl.org/pub/blfs/conglomeration/libatomic/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+  artifacts:
+    paths:
+      - openxt-workspace/
+    when: always
+    expire_in: 3 days
+  only:
+    variables:
+      - $BUILD_TYPE != ""
+
+.build_template: &build_definition  # Template for building images
+  <<: *pull-push-cache
+  image: ${EC2_DOCKER_REGISTRY}/build_image:debian_jessie
+  stage: build
+  before_script: *setup_cm
+  script:
+    - cd openxt-workspace
+    - DEPLOY_DIR=build-${CI_COMMIT_REF_NAME}/deploy
+    # Update the build/deploy directory with any cached contents, excluding IPK's
+    - if [ -d ${DEPLOY_DIR} ]; then rm -rf ${DEPLOY_DIR}/*; else mkdir -p ${DEPLOY_DIR}; fi
+    - mv ../deploy/* ${DEPLOY_DIR} || true
+    - rm -rf ${DEPLOY_DIR}/ipk || true
+    - if [[ "x$DEBUG" != "x" ]]; then ls -al build-${CI_COMMIT_REF_NAME}; ls -al ..; fi
+    # Update all the sources we're going to build, then build it
+    - repo sync --no-clone-bundle
+    - echo "openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} build \"$IMAGE_TO_BUILD\""
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} build "$IMAGE_TO_BUILD"
+  after_script:
+    - BUILD_DIR=openxt-workspace/build-${CI_COMMIT_REF_NAME}
+    # Ensure we have the log and deploy directories for caching
+    - mkdir -p {deploy,logs}
+    # Save the cooker logs, then get rid of the tmp* directories, they're too large
+    # Modify this if you need to debug your compile
+    - find ${BUILD_DIR}/tmp-glibc/log/cooker -name "20*" -exec cp {} logs \;
```

Comment:
This was a quick way of getting the logs for debugging artifacts. Since the `tmp-glibc` directory is removed each build, there should only ever be the current build there. Is there a version you'd prefer more?

Reply:
The logs under `"${TMPDIR}/log/cooker"` are all ending with `.log` and there's one symlink to the latest one. I would prefer the wildcard to be on this given that the date format could change (very unlikely).

Reply:
Updated it to copy all the directories in log/cooker/ since I was seeing two jobs with the same logfile name in different machine settings. I added the ability to save the failing jobs log file to the individual builds also, it'll only save the last one if there are multiple failures, but the entire builds output log will be saved regardless. The individual build jobs(vs. build all) save to logs/<job name>/ to make sure they're not going to clobber when archiving.